sycl/include/sycl/ext/oneapi/matrix/matrix-jit-use.hpp

```diff
@@ -0,0 +1,660 @@
+public:
+  __spv::__spirv_JointMatrixINTEL<
+      T, NumRows, NumCols, spv_matrix_layout_traits<Layout>::value,
+      spv_scope_traits<Group>::value, spv_matrix_use_traits<Use>::value> *spvm;
+  joint_matrix(Group sg) {
+#ifndef __SYCL_DEVICE_ONLY__
+    (void)sg;
+    throw runtime_error("joint matrix is not supported on host device.",
+                        PI_ERROR_INVALID_DEVICE);
+#endif // __SYCL_DEVICE_ONLY__
+  }
+
+  inline __SYCL_ALWAYS_INLINE wi_data<T, NumRows, NumCols, Use, Layout, Group>
+  get_wi_data() {
+    return wi_data<T, NumRows, NumCols, Use, Layout, Group>(*this);
+  }
+};
+
+template <typename Group, typename T, size_t NumRows, size_t NumCols, use Use,
+          layout Layout = layout::unused, access::address_space Space>
+inline __SYCL_ALWAYS_INLINE void
+joint_matrix_load(Group sg,
+                  joint_matrix<T, NumRows, NumCols, Use, Layout, Group> &res,
+                  multi_ptr<T, Space> src, size_t stride, layout MemL) {
+#ifdef __SYCL_DEVICE_ONLY__
+  T *Ptr = src.get();
+  switch (MemL) {
+  default:
+    assert(false && "Invalid Memory Layout!");
+  case layout::row_major:
+    res.spvm =
+        __spirv_JointMatrixLoadINTEL<T, NumRows, NumCols,
+                                     spv_matrix_use_traits<Use>::value,
+                                     spv_matrix_layout_traits<Layout>::value>(
+            Ptr, stride, __spv::MatrixLayout::RowMajor,
+            spv_scope_traits<Group>::value);
+    break;
+  case layout::col_major:
+    res.spvm =
+        __spirv_JointMatrixLoadINTEL<T, NumRows, NumCols,
+                                     spv_matrix_use_traits<Use>::value,
+                                     spv_matrix_layout_traits<Layout>::value>(
+            Ptr, stride, __spv::MatrixLayout::ColumnMajor,
+            spv_scope_traits<Group>::value);
+    break;
+  case layout::packed_a:
+    res.spvm =
+        __spirv_JointMatrixLoadINTEL<T, NumRows, NumCols,
+                                     spv_matrix_use_traits<Use>::value,
+                                     spv_matrix_layout_traits<Layout>::value>(
+            Ptr, stride, __spv::MatrixLayout::PackedA,
+            spv_scope_traits<Group>::value);
+    break;
+  case layout::packed_b:
+    res.spvm =
+        __spirv_JointMatrixLoadINTEL<T, NumRows, NumCols,
+                                     spv_matrix_use_traits<Use>::value,
+                                     spv_matrix_layout_traits<Layout>::value>(
+            Ptr, stride, __spv::MatrixLayout::PackedB,
+            spv_scope_traits<Group>::value);
+    break;
+  }
+#else
+  (void)sg;
+  (void)res;
+  (void)src;
+  (void)stride;
+  (void)MemL;
+  throw runtime_error("joint matrix is not supported on host device.",
+                      PI_ERROR_INVALID_DEVICE);
+#endif // __SYCL_DEVICE_ONLY__
+}
+
+template <typename Group, typename T, size_t NumRows, size_t NumCols, use Use,
+          layout MatL = layout::unused, access::address_space Space>
```

Comment:
Same situation here as with `joint_matrix_load`.  You can remove `layout MatL = layout::unused` from the template parameters then directly use layout::unused in `joint_matrix<T, NumRows, NumCols, Use, layout::unused, Group>`

Reply:
Otherwise it makes it legal for a user to try to call joint_matrix_load/store with a different layout from `layout::unused` which is wrong.